app/services/chemical_search_service.rb

```diff
@@ -62,8 +69,13 @@
   def cas_cleaned
     return unless @cas
 
-    if m = cas_number_regex.match(@cas)
-      m[2]
-    end
+    # Avoid DDOs by limiting the length of the CAS number
+    # e.g.
+    #   "cas " + " " * 1000000 + "12345-67-9"
+    #   "cas rn " + "blah " * 500000 + "12345-67-9"
+    #   "cas " + "123 " * 100000 + "12345-67-9" + " " * 1000000
+    @cas = @cas.to_s.strip.first(100)
+
+    CAS_NUMBER_REGEX.match(@cas).try(:[], 0)
```

Comment:
I've moved to:
- Using a non-capturing group, which matches the prefix but does not capture it as a separate group (saving on memory)
- Restricted the possible input size to 100 to avoid the chance of anyone manipulating the input

Reply:
I guess ideally, we'd do the stripping of the cas prefix in the frontend search - I'll open a new story for that